console/acl/acl.go

```diff
@@ -0,0 +1,392 @@
+		requiredPermissions = NewPermission(StorageData, PermissionRead)
+	case "/nakama.console.Console/ListSubscriptions":
+		requiredPermissions = NewPermission(InAppPurchase, PermissionRead)
+	case "/nakama.console.Console/ListUsers":
+		requiredPermissions = NewPermission(User, PermissionRead)
+	case "/nakama.console.Console/PromoteGroupMember":
+		requiredPermissions = NewPermission(Group, PermissionWrite)
+	case "/nakama.console.Console/RequireUserMfa":
+		requiredPermissions = NewPermission(User, PermissionWrite)
+	case "/nakama.console.Console/ResetUserMfa":
+		requiredPermissions = NewPermission(User, PermissionWrite)
+	case "/nakama.console.Console/UnbanAccount":
+		requiredPermissions = NewPermission(Account, PermissionWrite)
+	case "/nakama.console.Console/UnlinkApple":
+		requiredPermissions = NewPermission(Account, PermissionWrite)
+	case "/nakama.console.Console/UnlinkCustom":
+		requiredPermissions = NewPermission(Account, PermissionWrite)
+	case "/nakama.console.Console/UnlinkDevice":
+		requiredPermissions = NewPermission(Account, PermissionWrite)
+	case "/nakama.console.Console/UnlinkEmail":
+		requiredPermissions = NewPermission(Account, PermissionWrite)
+	case "/nakama.console.Console/UnlinkFacebook":
+		requiredPermissions = NewPermission(Account, PermissionWrite)
+	case "/nakama.console.Console/UnlinkFacebookInstantGame":
+		requiredPermissions = NewPermission(Account, PermissionWrite)
+	case "/nakama.console.Console/UnlinkGameCenter":
+		requiredPermissions = NewPermission(Account, PermissionWrite)
+	case "/nakama.console.Console/UnlinkGoogle":
+		requiredPermissions = NewPermission(Account, PermissionWrite)
+	case "/nakama.console.Console/UnlinkSteam":
+		requiredPermissions = NewPermission(Account, PermissionWrite)
+	case "/nakama.console.Console/UpdateAccount":
+		requiredPermissions = NewPermission(Account, PermissionWrite)
+	case "/nakama.console.Console/UpdateGroup":
+		requiredPermissions = NewPermission(Group, PermissionWrite)
+	case "/nakama.console.Console/UpdateSetting":
+		requiredPermissions = NewPermission(Settings, PermissionWrite)
+	case "/nakama.console.Console/WriteStorageObject":
+		requiredPermissions = NewPermission(StorageData, PermissionWrite)
+	case "/v2/console/storage/import":
+		// Special case for non-grpc gateway endpoint.
+		requiredPermissions = NewPermission(StorageDataImport, PermissionWrite)
+	default:
+		requiredPermissions = Admin
+	}
+
+	return userPermissions.HasAccess(requiredPermissions)
+}
+
+func New(acl map[string]*console.Permissions) Permission {
+	acc := None
+
+	for resource, permissions := range acl {
+		if permissions == nil {
+			continue
+		}
+
+		if permissions.Read {
+			acc = acc.Compose(NewPermissionFromString(resource, PermissionRead))
+		}
+
+		if permissions.Write {
+			acc = acc.Compose(NewPermissionFromString(resource, PermissionWrite))
+		}
+
+		if permissions.Delete {
+			acc = acc.Compose(NewPermissionFromString(resource, PermissionDelete))
+		}
+	}
+
+	return acc
```

Comment:
Remember to check the equivalence for the permissions received with the admin. The admin is someone with the set of bits all with 1. Unless you have all bits reserved for all resources, you will receive a subset bits only for declared resources and that number won't match the array with all bits with 1 but it will be equivalent.